### PR TITLE
Add Packer

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Add Packer for building custom vhd images https://www.packer.io/docs/builders/azure.html. Based on #3536

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
This is the second PR of a series of custom VHDs related PRs to breakdown #3536.
```

### CODE REVIEW

File: packer.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	@./packer/init-variables
+
+az-login:
+	az login --service-principal -u ${CLIENT_ID} -p ${CLIENT_SECRET} --tenant ${TENANT_ID}
+
+run-packer:
+	@packer version && make az-login && make init-packer && (make build-packer | tee packer-output)
+
+az-copy:
+	@make az-login && (azcopy --source "${OS_DISK_SAS}" --destination "${CLASSIC_BLOB}/${VHD_NAME}" --dest-sas "${CLASSIC_SAS_TOKEN}")
```
Comment (mboersma) [inline packer.mk]: I think `az login` should be a make target dependency instead:

```Makefile
az-copy: az-login
        azcopy --source "${OS_DISK_SAS}" --destination "${CLASSIC_BLOB}/${VHD_NAME}" --dest-sas "${CLASSIC_SAS_TOKEN}"
```

Same for the `generate-sas:` target below.

Comment (CecileRobertMichon) [reply packer.mk]: Good point, I'll fix it. Thanks!

File: packer.mk [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	@packer build -var-file=packer/settings.json packer/vhd-image-builder.json
+
+init-packer:
+	@./packer/init-variables
+
+az-login:
+	az login --service-principal -u ${CLIENT_ID} -p ${CLIENT_SECRET} --tenant ${TENANT_ID}
+
+run-packer: az-login
+	@packer version && make init-packer && (make build-packer | tee packer-output)
```
Comment (mboersma) [inline packer.mk]: When you invoke `make` from inside a Makefile, it should be referred to as `$(MAKE)` according to the [GNU docs](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable).

Comment (CecileRobertMichon) [reply packer.mk]: fixed

**Pull request merged**